Sources/StreamChatTestTools/DatabaseContainer_Mock.swift

```diff
@@ -0,0 +1,575 @@
+        try writeSynchronously { session in
+            let dto = NSEntityDescription
+                .insertNewObject(
+                    forEntityName: UserListQueryDTO.entityName,
+                    into: session as! NSManagedObjectContext
+                ) as! UserListQueryDTO
+            dto.filterHash = filter.filterHash
+            dto.filterJSONData = try JSONEncoder.default.encode(filter)
+        }
+    }
+    
+    func createMemberListQuery<ExtraData: UserExtraData>(query: _ChannelMemberListQuery<ExtraData>) throws {
+        try writeSynchronously { session in
+            try session.saveQuery(query)
+        }
+    }
+    
+    /// Synchronously creates a new MessageDTO in the DB with the given id.
+    func createMessage(
+        id: MessageId = .unique,
+        authorId: UserId = .unique,
+        cid: ChannelId = .unique,
+        text: String = .unique,
+        pinned: Bool = false,
+        pinnedByUserId: UserId? = nil,
+        pinnedAt: Date? = nil,
+        pinExpires: Date? = nil,
+        attachments: [AttachmentPayload] = [],
+        localState: LocalMessageState? = nil,
+        type: MessageType? = nil,
+        numberOfReplies: Int = 0
+    ) throws {
+        try writeSynchronously { session in
+            try session.saveChannel(payload: XCTestCase().dummyPayload(with: cid))
+            
+            let message: MessagePayload<NoExtraData> = .dummy(
+                type: type,
+                messageId: id,
+                attachments: attachments,
+                authorUserId: authorId,
+                text: text,
+                pinned: pinned,
+                pinnedByUserId: pinnedByUserId,
+                pinnedAt: pinnedAt,
+                pinExpires: pinExpires
+            )
+            
+            let messageDTO = try session.saveMessage(payload: message, for: cid)
+            messageDTO.localMessageState = localState
+            
+            for idx in 0..<numberOfReplies {
+                let reply: MessagePayload<NoExtraData> = .dummy(
+                    type: .reply,
+                    messageId: .unique,
+                    parentId: id,
+                    authorUserId: authorId,
+                    text: "Reply \(idx)"
+                )
+                
+                let replyDTO = try session.saveMessage(payload: reply, for: cid)
+                messageDTO.replies.insert(replyDTO)
+            }
+        }
+    }
+    
+    func createMember(
+        userId: UserId = .unique,
+        role: MemberRole = .member,
+        cid: ChannelId,
+        query: ChannelMemberListQuery? = nil
+    ) throws {
+        try writeSynchronously { session in
+            try session.saveMember(
+                payload: .dummy(userId: userId, role: role),
+                channelId: query?.cid ?? cid,
+                query: query
+            )
+        }
+    }
+}
+
+extension XCTestCase {
+    static let channelCreatedDate = Date.unique
```

Comment:
The file is named `DatabaseContainer_Mock` but theres 2x more LOC with an unrelated `XCTest` extension ;-)

Reply:
oops, good point, I'll refactor to its own file